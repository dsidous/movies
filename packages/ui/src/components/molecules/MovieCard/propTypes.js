import { string, object } from 'prop-types';

export const propTypes = {
  img_base_path: string,
  movie: object.isRequired,
  media: string,
};

export const defaultProps = {};
