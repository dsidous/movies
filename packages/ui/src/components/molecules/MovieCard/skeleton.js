import React from 'react';
import { Link } from 'react-router-dom';
import Skeleton from '@material-ui/lab/Skeleton';

import useStyles from './styles';

export default () => {
  const classes = useStyles();
  return (
    <li className={classes.root} style={{ minHeight: '230px' }}>
      <Link to=".">
        <Skeleton
          className={classes.poster}
          variant="rect"
          style={{ backgroundColor: '#999' }}
        />
      </Link>
      <div>
        <Skeleton width="70%" />
      </div>
      <div>
        <Skeleton width={2} />
      </div>
      <div>
        <Skeleton width="90%" />
        <Skeleton width="90%" />
      </div>
    </li>
  );
};
