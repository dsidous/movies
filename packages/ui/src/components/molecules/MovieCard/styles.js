import { makeStyles } from '@material-ui/core/styles';

export default makeStyles(
  theme => ({
    root: {
      listStyle: 'none',
      background: theme.cardBg,
      borderRadius: 8,
      margin: '10px 0 20px',
      overflow: 'hidden',
      position: 'relative',
      boxShadow: '0 0 30px 10px rgba(107, 106, 106, 0.1)',

      [theme.breakpoints.up('sm')]: {
        display: 'grid',
        gridColumnGap: 10,
        gridTemplateRows: '40px 40px',
        margin: 0,
        padding: 0,
      },

      [theme.breakpoints.up('md')]: {
        display: 'grid',
        gridTemplateColumns: 'minmax(115px, 1fr) minmax(200px, 2fr)',
        gridColumnGap: 10,
        gridTemplateRows: '40px 40px',
        padding: 0,
      },

      '& a': {
        [theme.breakpoints.up('sm')]: {
          gridRow: '1 / 4',
        },
      },
    },

    poster: {
      cursor: 'pointer',
      display: 'block',
      height: '100%',
      width: '100%',
    },

    noImage: {
      background: '#666',
      color: '#444',
      cursor: 'pointer',
      font: '65px/240px FontAwesome',
      height: 240,
      textAlign: 'center',
      width: 165,

      '&:before': {
        content: '"\f03e"',
      },
    },

    title: {
      display: 'block',
      float: 'left',
      fontSize: 20,
      margin: '10px 0 0 10px',
      overflow: 'hidden',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      width: '70%',
    },

    rating: {
      display: 'block',
      float: 'right',
      fontSize: 20,
      margin: '10px 10px 0 0',
      paddingLeft: 22,
      position: 'relative',
      textAlign: 'right',

      '&:before': {
        font: '20px/20px FontAwesome',
        color: '#666',
        content: '"\\f005"',
        left: 0,
        position: 'absolute',
      },
    },

    year: {
      float: 'left',
      margin: '5px 0 0 10px',
      paddingLeft: 14,
      position: 'relative',

      '&:before': {
        content: '"\\f073"',
        left: 0,
        font: '12px/12px FontAwesome',
        position: 'absolute',
        top: 3,
      },
    },

    genres: {
      display: 'block',
      float: 'right',
      margin: '5px 10px 10px 0',
      overflow: 'hidden',
      textAlign: 'right',
      fontStyle: 'italic',
      textOverflow: 'ellipsis',
      whiteSpace: 'nowrap',
      width: '70%',
    },

    overview: {
      display: 'none',

      [theme.breakpoints.up('sm')]: {
        display: 'block',
        margin: '20px 10px',
      },

      '&.long::after': {
        content: '"..."',
      },
    },
  }),
  { name: 'showCard' },
);
