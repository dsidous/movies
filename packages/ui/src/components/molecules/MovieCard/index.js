/* eslint-disable camelcase */
import React from 'react';
import { Link } from 'react-router-dom';

import { propTypes } from './propTypes';
import NoImage from '../../../images/noimage.jpg';
import NoBdImage from '../../../images/nobdimage.jpg';
import WatchlistBookmark from '../../atoms/WatchlistBookmark';

const MovieCard = ({ img_base_path, movie, media }) => {
  const {
    id,
    title,
    overview,
    poster_path,
    backdrop_path,
    vote_average,
    release_date,
    name,
    first_air_date,
  } = movie;

  const mtitle = title || name;
  const date = release_date || first_air_date;
  const imagePath =
    poster_path !== null
      ? [img_base_path + poster_path, img_base_path + backdrop_path]
      : [NoImage, NoBdImage];

  return (
    <li className="movies-list__element">
      <Link to={`/${media}/${id}`}>
        <picture className="movies-poster">
          <source media="(min-width: 941px)" srcSet={imagePath[0]} />
          <source media="(max-width: 940px)" srcSet={imagePath[1]} />
          <img src={imagePath[0]} className="movies-poster" alt={mtitle} />
        </picture>
      </Link>
      <div className="clearfix">
        <div className="movies-title">{mtitle}</div>
        <div className="movies-rating">{vote_average}</div>
      </div>
      <div className="clearfix">
        <div className="movies-year">{date.slice(0, 4)}</div>
      </div>
      <div
        className={[
          'movies-overview ',
          overview.length > 200 ? 'long' : '',
        ].join(' ')}
      >
        {overview.slice(0, overview.indexOf(' ', 200))}
      </div>
      <WatchlistBookmark movie={movie} />
    </li>
  );
};

MovieCard.propTypes = propTypes;

export default MovieCard;
