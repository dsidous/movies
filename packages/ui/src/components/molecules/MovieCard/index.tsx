import React from 'react';
import { Link } from 'react-router-dom';
import { Box, Typography } from '@material-ui/core';
import { useTheme } from '@material-ui/core/styles';

import { WatchlistBookmark } from '@atoms/index';
import useConfig from '@hooks/useConfig';
import Skeleton from './skeleton';
import useStyles from './styles';
import { Show } from '@typesRoots/show';

interface Props {
  movie?: Show;
  media?: string;
}

const MovieCard: React.FC<Props> = ({ movie, media }) => {
  if (!movie) {
    return <Skeleton />;
  }

  const { getImageURL } = useConfig();
  const classes = useStyles();
  const theme = useTheme();

  const {
    id,
    title,
    overview,
    poster_path,
    backdrop_path,
    vote_average,
    release_date,
    name,
    first_air_date,
  } = movie;

  const mtitle = title || name;
  const date = release_date || first_air_date;
  const posterURL = getImageURL({
    filePath: poster_path,
    mediaType: 'poster',
    size: 1,
  });

  const backdropURL = getImageURL({
    filePath: backdrop_path,
    mediaType: 'backdrop',
    size: 1,
  });

  return (
    <li className={classes.root}>
      <Link to={`/${media}/${id}`}>
        <picture className={classes.poster}>
          <source
            media={`(min-width: ${theme.breakpoints.values.md}px)`}
            srcSet={posterURL}
          />
          <source
            media={`(max-width: ${theme.breakpoints.values.md}px)`}
            srcSet={backdropURL}
          />
          <img className={classes.poster} src={posterURL} alt={mtitle} />
        </picture>
      </Link>
      <Box display='flex' justifyContent='space-between' alignItems='flex-end'>
        <Typography variant='h5' className={classes.title}>
          {mtitle}
        </Typography>
        <Typography className={classes.rating}>{vote_average}</Typography>
      </Box>
      <Typography variant='subtitle1' className={classes.year}>
        {date.slice(0, 4)}
      </Typography>
      <Typography
        variant='body2'
        className={[classes.overview, overview.length > 200 ? 'long' : ''].join(
          ' '
        )}
      >
        {overview.slice(0, overview.indexOf(' ', 200))}
      </Typography>
      <WatchlistBookmark movie={movie} />
    </li>
  );
};

export default MovieCard;
