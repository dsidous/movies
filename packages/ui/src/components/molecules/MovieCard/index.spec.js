import React from 'react';
import { shallow } from 'enzyme';
import MovieCard from '.';

const mockProps = {
  movie: {
    id: 1,
    backdrop_path: 'path/to/image',
    poster_path: 'path/to/image',
    genres: [
      {
        id: 1,
        genre_name: 'genre',
      },
    ],
    name: 'name',
    release_date: '01.01.2010',
    vote_average: 1,
    overview: 'overview',
  },
  media: 'movie',
  img_base_path: 'path/to/image',
};

describe('Molecules/MovieCard', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MovieCard {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
