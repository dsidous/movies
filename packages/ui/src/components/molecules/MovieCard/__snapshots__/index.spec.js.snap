// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules/MovieCard should render as expected 1`] = `ShallowWrapper {}`;
